lms/djangoapps/grades/api/v1/views.py

```diff
@@ -0,0 +1,233 @@
+        is successful, an HTTP 200 "OK" response is returned.
+        The HTTP 200 response has the following values.
+        * username: A string representation of a user's username passed in the request.
+        * course_id: A string representation of a Course ID.
+        * passed: Boolean representing whether the course has been
+                  passed according the course's grading policy.
+        * percent: A float representing the overall grade for the course
+        * letter_grade: A letter grade as defined in grading_policy (e.g. 'A' 'B' 'C' for 6.002x) or None
+    **Example GET Response**
+        [{
+            "username": "bob",
+            "course_key": "course-v1:edX+DemoX+Demo_Course",
+            "passed": false,
+            "percent": 0.03,
+            "letter_grade": null,
+        },
+        {
+            "username": "fred",
+            "course_key": "course-v1:edX+DemoX+Demo_Course",
+            "passed": true,
+            "percent": 0.83,
+            "letter_grade": "B",
+        },
+        {
+            "username": "kate",
+            "course_key": "course-v1:edX+DemoX+Demo_Course",
+            "passed": false,
+            "percent": 0.19,
+            "letter_grade": null,
+        }]
+    """
+    def get(self, request, course_id=None):
+        """
+        Gets a course progress status.
+        Args:
+            request (Request): Django request object.
+            course_id (string): URI element specifying the course location.
+        Return:
+            A JSON serialized representation of the requesting user's current grade status.
+        """
+        use_email = str_to_bool(request.GET.get('use_email'))
+        username = request.GET.get('username')
+
+        if not course_id:
+            course_id = request.GET.get('course_id')
+
+        if username:
+            access_action = 'load'
+        else:
+            access_action = 'staff'
+
+        course = self._get_course(course_id, request.user, access_action)
+        if isinstance(course, Response):
+            # Returns a 404 if course_id is invalid, or request.user is not enrolled in the course
+            return course
+
+        if username:
+            grade_user = self._get_effective_user(request, course)
+            if isinstance(grade_user, Response):
+                # Returns a 403 if the request.user can't access grades for the requested user,
+                # or a 404 if the requested user does not exist.
+                return grade_user
+
+            course_grade = CourseGradeFactory().read(grade_user, course)
+
+            return Response([
+                self._make_grade_response(grade_user, course, course_grade, use_email)
+            ])
+        else:
+            enrollments_in_course = enrollment_data.get_user_enrollments(
+                course.id, serialize=False
+            )
+
+            if not enrollments_in_course:
+                return self.make_error_response(
```

Comment:
Why are no enrollments considered a `404`?  It should just be an empty list that's returned, right?